### PR TITLE
Migrate from assertThat to assertQueryFailure

### PR DESCRIPTION
**Pull request by pvliborio**
Solves #8274 

- Migrate from io.trino.tempto.assertions.QueryAssert#assertThat(QueryCallback) to assertQueryFailure(QueryCallback)

### CODE REVIEW

**Review by hashhar — COMMENTED** [review]
Please take a look at Checkstyle errors.

Also use `hasMessageContaining` instead of `hasMessage`. The `failsWithMessage` checked using `contains` instead of equality.

**Review by pvliborio — COMMENTED** [review]
The .hasMessageMatching doesn't match `java.sql.SQLException: `. Would it be okay to remove that part?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java:103]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java:1424]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java:1395]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroWritesSmokeTest.java
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroWritesSmokeTest.java:70]: should be safe to remove `java.sql.SQLException`.

**Review by hashhar — APPROVED** [review]
LGTM % ci failures.

It's surprising that the earlier `failsMessageMatching` passed for regexes having exception name in the assertion. AFAIK exception's `message` doesn't include the exception name.

cc: @findepi Do you have any idea why this is behaving like this? The old and new impl look same to me (`str.matches(regex)`)?

**Comment by pvliborio** [comment]
@hashhar @findepi I didn't understand the error in CI. Can you help me?

**Comment by hashhar** [comment]
> @hashhar @findepi I didn't understand the error in CI. Can you help me?

Can you push an empty commit? That failure looks like a flakyness.

`git commit --allow-empty -m 'empty for ci'`

**Comment by pvliborio** [comment]
> > @hashhar @findepi I didn't understand the error in CI. Can you help me?
> 
> Can you push an empty commit? That failure looks like a flakyness.
> 
> `git commit --allow-empty -m 'empty for ci'`

Done.

**Comment by pvliborio** [comment]
> > @hashhar @findepi I didn't understand the error in CI. Can you help me?
> 
> Can you push an empty commit? That failure looks like a flakyness.
> 
> `git commit --allow-empty -m 'empty for ci'`

Worked! Thank you.

**Comment by hashhar** [comment]
Thanks @pvliborio . All tests passed.
Have you sent the signed CLA using the same email you use with your GitHub account/the one you have in your GitHub commit? We require the CLA before we can merge the code.

**Comment by pvliborio** [comment]
> Thanks @pvliborio . All tests passed.
> Have you sent the signed CLA using the same email you use with your GitHub account/the one you have in your GitHub commit? We require the CLA before we can merge the code.

Yes. Sent on 06/21/2021 08:57:34 -0300 to cla@trino.io.

**Comment by hashhar** [comment]
@martint Can you please help with the CLA here? This looks good to go otherwise.

**Comment by pvliborio** [comment]
@hashhar CLA done!

**Comment by hashhar** [comment]
@clabot check

**Comment by hashhar** [comment]
@cla-bot check

**Comment by pvliborio** [comment]
@hashhar I think this resolved, sorry, it was my mistake in the first commit.

**Comment by hashhar** [comment]
Awesome @pvliborio. Can you please squash all the commits into a single one. This is logically a single change so should be kept as such in the history.

Will merge once CI finishes after you push.

**Comment by pvliborio** [comment]
@hashhar Done. Thanks for the help!

**Review by hashhar — APPROVED** [review]

**Pull request merged**
